src/ray/core_worker/reference_count.cc

```diff
@@ -1167,29 +1188,36 @@ size_t ReferenceCounter::GetObjectSize(const ObjectID &object_id) const {
 
 bool ReferenceCounter::HandleObjectSpilled(const ObjectID &object_id,
                                            const std::string spilled_url,
-                                           const NodeID &spilled_node_id, int64_t size,
-                                           bool release) {
+                                           const NodeID &spilled_node_id, int64_t size) {
   absl::MutexLock lock(&mutex_);
   auto it = object_id_refs_.find(object_id);
   if (it == object_id_refs_.end()) {
     RAY_LOG(WARNING) << "Spilled object " << object_id << " already out of scope";
     return false;
   }
+  if (it->second.OutOfScope(lineage_pinning_enabled_)) {
```

Comment:
This seems to be unrelated to this PR. Is it a bug you discovered or sth?

Reply:
Actually it is related. We shouldn't add the spilled location if the object is already out of scope because it means the spilled copy is already being deleted by the raylet. I'll add a note to explain this.